src/App.js

```diff
@@ -658,6 +658,20 @@ export class App extends React.Component<AppProps, AppState> {
                                 this.handleProgramSequenceChange(newProgramSequence);
                             }
                             break;
+                        case("deleteCurrentStep"):
+                            const currentElement = document.activeElement;
+                            // $FlowFixMe: Not all elements have dataset property
+                            const index = parseInt(currentElement.dataset.stepnumber);
```

Comment:
Add-nodes have a `stepnumber` data attribute also. So this test isn't enough to safely determine if the active element is a program step. Let's add a new data attribute that specifies the element type called "`data-controltype`". For program steps, use:

    data-controltype='programStep'

And for add-nodes use:

    data-controltype=`addNode`

And then in this `deleteCurrentStep` handler, check to see if the `activeElement` has a data attribute `controltype` equal to `programStep`.